src/api/app/controllers/webui/distributions_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Webui::DistributionsController < Webui::WebuiController
+  before_action :set_project
+  after_action :verify_authorized
+
+  # GET /projects/:project/distributions/new
+  def new
+    authorize @project, :update?
+
+    @distributions = Distribution.all.group_by(&:vendor)
+    return if @distributions.present?
+
+    if User.admin_session?
+      redirect_to(new_interconnect_path,
+                  alert: 'There are no distributions configured. Maybe you want to connect to one of the public OBS instances?')
+    else
+      redirect_to(project_repositories_path(project: @project))
+    end
+  end
+
+  # PATCH /projects/:project_name/distributions/:id/toggle
+  def toggle
+    authorize @project, :update?
+    @distribution = Distribution.find(params[:distribution])
+
+    @repository = @project.repositories.find_by(name: @distribution.reponame)
+    if @project.has_distribution(@distribution.project, @distribution.repository)
+      destroy_repository
+    else
+      create_repository_from_distribution
+    end
+  end
+
+  private
+
+  def create_repository_from_distribution
+    repository = Repository.new_from_distribution(@distribution)
+    repository.project = @project
+
+    if repository.save
+      @project.store(comment: "Added #{repository.name} repository")
+    else
+      flash.now[:error] = "Can't add repository: #{repository.errors.full_messages.to_sentence}"
```

Comment:
When trying to add a variant when there is already another enabled (like enabling `Arch Extra`, then trying to enable `Arch Community`), this results in an error message like:

```
Can't add repository: Name Arch is already used by a repository of this project 
```

This is somewhat cryptic. Could we improve it somehow?

Reply:
how is this cryptic?

Reply:
I think an error message like `Only one variant of repository_name can be enabled.` would be much better.

Reply:
This is not limited to variants. You can't have two repositories with the same name no matter what their relationship is.

Reply:
Hm maybe I could catch exactly this validation error and change the message in the distribution controller...

Reply:
Might also be related to #11317

Reply:
will you investigate if it is related with #11317 and try to address it here?

Reply:
No I will address this later.